### PR TITLE
No salt for Android

### PR DESCRIPTION
**Pull request by lnkuiper**
Title

### CODE REVIEW

File: src/include/duckdb/execution/ht_entry.hpp [hunk 1]
```diff
@@ -21,10 +21,16 @@ namespace duckdb {
 */
 struct ht_entry_t { // NOLINT
 public:
+#if defined(__ANDROID__) || defined(ANDROID)
```
Comment (szarnyasg) [inline src/include/duckdb/execution/ht_entry.hpp:24]:
```suggestion
#if defined(__ANDROID__)
```

This should be sufficient, the codebase doesn't use the `ANDROID` macro name.

File: src/include/duckdb/execution/ht_entry.hpp [hunk 2]
```diff
@@ -13,6 +13,11 @@
 
 namespace duckdb {
 
+// #if !defined(DISABLE_POINTER_SALT) && defined(__ANDROID__)
+// Google, why does Android need 18446744 TB of address space?
```
[the remark this answers is not in the document]
Comment (lnkuiper) [reply src/include/duckdb/execution/ht_entry.hpp:17]: This was a lot of hacking around to try and get Android working before presenting the results yesterday. I've done a better refactor now and added a CI run that verifies that we have correct results without salt.

**Review by szarnyasg — APPROVED** [review]

**Comment by lnkuiper** [comment]
This PR added a flag to a NightlyTests run, so that's why all that got triggered. NightlyTests is currently broken, but I think this PR is ready to go

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**